src/AbstractFactory/ReflectionBasedAbstractFactory.php

```diff
@@ -106,7 +106,9 @@ public function __construct(array $aliases = [])
     /**
      * {@inheritDoc}
      *
-     * @return DispatchableInterface
+     * @param class-string<T>|string $requestedName
+     * @return ($requestedName is class-string<T> ? T : object)
+     * @template T as object
```

Comment:
Shouldn't it be 'of object' or does 'as object' do the same thing?

Reply:
Something here is not right. I've seen this before. Template constraint has no effect.

Reply:
Look at this https://github.com/laminas/laminas-inputfilter/pull/97